src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -0,0 +1,322 @@
+                'token' => $token,
+                'expires_at' => $this->createDateTime($now + $this->initialTtl),
+            ),
+        );
+
+        $options = array(
+            'upsert' => true,
+        );
+
+        $key->reduceLifetime($this->initialTtl);
+
+        try {
+            $this->getCollection()->updateOne($filter, $update, $options);
+        } catch (WriteException $e) {
+            throw new LockConflictedException('Failed to acquire lock', 0, $e);
+        }
+
+        if ($key->isExpired()) {
+            throw new LockExpiredException(sprintf('Failed to store the "%s" lock.', $key));
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function waitAndSave(Key $key)
+    {
+        throw new NotSupportedException(sprintf(
+            'The store "%s" does not supports blocking locks.',
+            __CLASS__
+        ));
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws LockStorageException
+     * @throws LockExpiredException
+     */
+    public function putOffExpiration(Key $key, $ttl)
+    {
+        $now = microtime(true);
+
+        $filter = array(
+            '_id' => (string) $key,
+            'token' => $this->getUniqueToken($key),
+            'expires_at' => array(
+                '$gte' => $this->createDateTime($now),
+            ),
+        );
+
+        $update = array(
+            '$set' => array(
+                '_id' => (string) $key,
+                'expires_at' => $this->createDateTime($now + $ttl),
+            ),
+        );
+
+        $options = array(
+            'upsert' => true,
+        );
+
+        $key->reduceLifetime($ttl);
+
+        try {
+            $this->getCollection()->updateOne($filter, $update, $options);
+        } catch (WriteException $e) {
+            $writeErrors = $e->getWriteResult()->getWriteErrors();
+            if (1 === \count($writeErrors)) {
+                $code = $writeErrors[0]->getCode();
+            } else {
+                $code = $e->getCode();
+            }
+            // Mongo error E11000 - DuplicateKey
+            if (11000 === $code) {
+                throw new LockConflictedException('Failed to put off the expiration of the lock', 0, $e);
+            }
+            throw new LockStorageException($e->getMessage(), 0, $e);
+        }
+
+        if ($key->isExpired()) {
+            throw new LockExpiredException(sprintf(
+                'Failed to put off the expiration of the "%s" lock within the specified time.',
+                $key
+            ));
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete(Key $key)
+    {
+        $filter = array(
+            '_id' => (string) $key,
+            'token' => $this->getUniqueToken($key),
+        );
+
+        $options = array();
+
+        $this->getCollection()->deleteOne($filter, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function exists(Key $key)
```

Comment:
`exists(Key $key): bool`

Reply:
added, initially I wasn't sure if I should be doing this kind of stuff since the parent doesn't do it... but it seems to work.